torch/jit/__init__.py

```diff
@@ -19,6 +19,33 @@
 _unflatten = torch._C._jit_unflatten
 
 
+# This global variable is set when we are tracing a *forwards* computation.
+# It is intended to be a cheap way to test if tracing has occurred, before
+# doing the slower path using `get_tracing_state` (below.)
+_tracing = False
+
+
+def get_tracing_state(*args):
+    if not torch._C._is_tracing(args):
+        return None
+    return torch._C._get_tracing_state(args)
+
+
+@contextlib.contextmanager
+def scope(scope_name, vars):
+    if not torch._C._is_tracing(vars):
+        tracing_state = None
+    else:
+        tracing_state = torch._C._get_tracing_state(vars)
```

Comment:
It seems like you duplicated the check above.

Reply:
Fixed